halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/HalconfigParser.java

```diff
@@ -104,8 +100,11 @@ public Halconfig setInmemoryHalConfig(ByteArrayInputStream is) throws IllegalArg
   }
 
   private InputStream getHalconfigStream() throws FileNotFoundException {
-    String path = useBackup ? backupHalconfigPath : halconfigPath;
-    return new FileInputStream(new File(path));
+    String baseDirectory =
+        useBackup
+            ? halconfigDirectoryStructure.getBackupConfigPath().toString()
+            : halconfigDirectoryStructure.getHalconfigDirectory();
+    return new FileInputStream(new File(baseDirectory, "config"));
```

Comment:
`halconfigDirectoryStructure.getBackupConfigPath()` already returns the path including `config` file, leading to a double "config" in case of backups: https://github.com/spinnaker/halyard/blob/master/halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/HalconfigDirectoryStructure.java#L109

Reply:
Good catch! Updated.